crates/wdk/src/print.rs

```diff
@@ -153,16 +153,16 @@ mod dbg_print_buf_writer {
     }
 
     impl fmt::Write for DbgPrintBufWriter {
+        // Traverses the string and writes all non-null bytes to the buffer.
+        // If the buffer is full, flushes the buffer and continues writing.
+        // Finishes with a non-flushed buffer containing the last
+        // non-null bytes of the string.
         fn write_str(&mut self, s: &str) -> fmt::Result {
             let mut str_byte_slice = s.as_bytes();
             let mut remaining_buffer = &mut self.buffer[self.used..Self::USABLE_BUFFER_SIZE];
             let mut remaining_buffer_len = remaining_buffer.len();
 
-            if let Some(first_non_null_byte_pos) = str_byte_slice.iter().position(|&b| b != b'\0') {
-                str_byte_slice = &str_byte_slice[first_non_null_byte_pos..];
-            } else {
-                return Ok(());
-            }
+            str_byte_slice = advance_slice_to_next_non_null_byte(str_byte_slice);
```

Comment:
Before the refactoring we were returning from `write_str` here if we could not find any non-null char. Perhaps we should retain that behaviour.